### PR TITLE
Internal signal refactors

### PR DESCRIPTION
**Pull request by crisbeto**
Reorganizes some signal-related code to make it easier to reuse.

### CODE REVIEW

File: packages/compiler-cli/src/transformers/jit_transforms/signal_metadata_transform_factory.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import {isAngularDecorator} from '../../ngtsc/annotations';
+import {Decorator, ReflectionHost} from '../../ngtsc/reflection';
+import {addImports} from '../../ngtsc/transform';
+import {ImportManager} from '../../ngtsc/translator';
+
+/** Decorators for classes that should be transformed. */
+const decoratorsWithInputs = ['Directive', 'Component'];
+
+/** Function that can be used to transform a class property to add signal-specific metadata. */
```
Comment (devversion) [inline packages/compiler-cli/src/transformers/jit_transforms/signal_metadata_transform_factory.ts:19]: Nit: Would be great to rename this to something for "initializer-based APIs` as output for example will not use signals, and needs a similar transform.

**Review by devversion — APPROVED** [review]

**Comment by devversion** [comment]
Note: Non-public facing change of the JIT transform, so we can skip the required explicit reviews from tooling side.

**Review by alan-agius4 — APPROVED** [review]
Reviewed of #tooling-cli-shared-api

**Review by crisbeto — COMMENTED** [review]
I can't approve my own PR, but LGTM on Paul's changes.

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit d74ee6e343ef319acb33d5e05bf96b3150050266.

**Pull request closed without merging**